datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -904,6 +977,7 @@ impl BinaryExpr {
         scalar: &ScalarValue,
     ) -> Result<Option<Result<ArrayRef>>> {
         let bool_type = &DataType::Boolean;
+        let left_type = array.data_type();
```

Comment:
Is there any reason to pass in `left_type` to the `binary_primitive_array_op_dyn_scalar!` (as it already gets `array` maybe it could call `array.data_type()` directly when needed)?

Not that I see anything wrong with this, but I was curious

Reply:
You're correct. `left_type` is not necessary to be passed to `binary_primitive_array_op_dyn_scalar!`. Let me remove it.